src/python/pants/invalidation/cache_manager.py

```diff
@@ -172,20 +172,34 @@ def __init__(self, cache_manager, target, cache_key):
     super(VersionedTarget, self).__init__(cache_manager, [self])
     self.id = target.id
 
+  def _results_dir_prefix(self, root_dir):
```

Comment:
How about add a root_dir or results_prefix kwarg to the cache_manager constructor?

Task.py could pass self.workdir as a kwarg, and would break no current callers. Then the cache_manager could just hold the reference to the prefix as a property or something.

I got ~ 95% done with a refactor that added this but decided to stop since I wasn't confident it would be accepted into master: https://github.com/mateor/pants/blob/b77f27644e06ff7ef3753753f8bacfea83c17d1f/src/python/pants/invalidation/cache_manager.py#L268